fleet_adapter_mir_actions/fleet_adapter_mir_actions/rmf_wait_until.py

```diff
@@ -0,0 +1,485 @@
+            )
+            return True
+
+        # Log the robot waiting every X seconds
+        seconds_passed = round(now - self.start_time)
+        if seconds_passed % self.update_gap == 0:
+            self.context.node.get_logger().info(
+                f'{seconds_passed} seconds have passed since robot '
+                f'[{self.context.name}] started its waiting action.'
+            )
+
+        return False
+
+    def create_move_off_cb(self, description: dict, custom_modules: dict):
+        signal_cb = None
+        signal_type = None
+
+        # Determine which move off signal config to use. Any config populated
+        # in the task description overrides the default config provided in
+        # action config.
+        if 'signal_type' in description:
+            signal_type = description['signal_type']
+        elif self.default_signal is not None:
+            signal_type = self.default_signal
+        else:
+            # There is no move off signal provided, we will just wait for the
+            # duration of the configured timeout
+            timeout = description.get('timeout', self.wait_timeout)
+            self.context.node.get_logger().info(
+                f'No move off signal was configured for [{self.context.name}]'
+                f', the robot will begin waiting until the configured timeout '
+                f'of [{timeout}] seconds.'
+            )
+            signal_cb = lambda: False
+            return signal_cb
+
+        signal_config = None
+        if signal_type in self.signal_config:
+            # Signal is preconfigured in the fleet action config
+            signal_config = self.signal_config[signal_type]
+            signal_type = signal_config['signal_type']
+        else:
+            signal_config = description.get('signal_config')
+        if signal_config is None:
+            self.context.node.get_logger().error(
+                f'The submitted signal type/config is invalid for '
+                f'{signal_type}!'
+            )
+            return None
+
+        match signal_type:
+            case 'mission':
+                mission_name = signal_config['mission_name']
+                resubmit_on_abort = signal_config.get(
+                    'resubmit_on_abort', False)
+                retry_count = signal_config.get('retry_count', 10)
+                signal_cb = self.create_mission_move_off_cb(
+                    mission_name, retry_count, resubmit_on_abort)
+            case 'plc':
+                register = signal_config['register']
+                self.context.node.get_logger().info(
+                    f'Configuring robot [{self.context.name}] move off '
+                    f'signal: robot will wait until the PLC register '
+                    f'{register} returns True.'
+                )
+                signal_cb = lambda: self.check_plc_register(register)
+            case 'custom':
+                module = custom_modules.get(signal_type)
```

Comment:
Looks like this line is always equivalent to 

```python
module = custom_modules.get('custom')
```

which doesn't seem right to me. I think this wants to be

```python
module = custom_modules.get(signal_config['module'])
```